pkg/blockbuilder/blockbuilder.go

```diff
@@ -331,6 +330,12 @@ func (b *BlockBuilder) consumePartitions(ctx context.Context, part int32, mark t
 		fetches.EachPartition(func(ftp kgo.FetchTopicPartition) {
 			level.Info(b.logger).Log("msg", "consumed", "part", ftp.Partition, "hi", ftp.HighWatermark, "lo", ftp.LogStartOffset, "batch_size", len(ftp.Records))
 
+			if ftp.Err != nil {
+				level.Error(b.logger).Log("msg", "failed to fetch records", "part", ftp.Partition, "err", ftp.Err)
+				// TODO(codesome): add metric?
+				return
+			}
```

Comment:
I believe this chunk is redundant, as we already handle (log) these errors above, when we iterate over the `fetches.EachError`.